worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/XmlConverter.java

```diff
@@ -61,7 +79,7 @@ private XmlConverter()
         for (Map.Entry<String, ReferencedData> referencedDataEntry : fields) {
             // Get the field name and value
             final String fieldName = referencedDataEntry.getKey();
-            final String fieldValue = getContentAsStringEx(dataSource, referencedDataEntry);
+            String fieldValue = getContentAsStringEx(dataSource, referencedDataEntry);
```

Comment:
I thought that you will just update the fieldValue here if it's a CONTENT field with headers and let the code proceed as normal... especially that there's some post processing on it - removeInvalidXmlElementNameChars & removeInvalidXmlElementTextChar
Maybe I'm missing something...

Reply:
As discussed, benefit of updating the value after the loop here is that all fields will be available as string values due to the loop, avoiding going to data store twice potentially. If done inside the for loop we couldn't guarantee all the required fields were converted at that point.

I had intended to change it there until thought of that downside which is why I had removed the 'final' modifier. Added that back in.